File: android_webview/javatests/src/org/chromium/android_webview/test/util/RendererProcessMetricsProviderUtils.java
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.android_webview.test.util;

import org.jni_zero.JNINamespace;
import org.jni_zero.NativeMethods;

/** Utilities for force recording renderer process metrics. */
@JNINamespace("android_webview")
public class RendererProcessMetricsProviderUtils {
    @NativeMethods
    public interface Natives {
        /** Calls to RendererProcessMetricsProvider to force record histograms. */
        void forceRecordHistograms();
    }
}

File: android_webview/test/browser/renderer_process_metrics_provider_utils.cc
// Copyright 2021 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.


#include "android_webview/browser/metrics/renderer_process_metrics_provider.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "android_webview/test/webview_instrumentation_test_native_jni/RendererProcessMetricsProviderUtils_jni.h"

namespace android_webview {

// static
void JNI_RendererProcessMetricsProviderUtils_ForceRecordHistograms(
    JNIEnv* env) {
  RendererProcessMetricsProvider().ProvideCurrentSessionData(nullptr);
}

}  // namespace android_webview
